src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -1167,6 +1167,13 @@ void CompositeClosure::compute_cdf()
     const size_t closure_count = get_num_closures();
     if (closure_count > 0)
     {
+        if (closure_count == 1)
```

Comment:
No, because foundation::CDF allocates memory.
Maybe we could refactor foundation::CDF to remove the container part out of it?